2016-2017_Starstruck/Module Test Code/FileIO_Test/src/AutonRecorder.c

```diff
@@ -0,0 +1,193 @@
+    root = NULLFRAME;
+    fileName = "";
+  }
+
+  void stopRecording(){
+    taskDelete(recordTask);
+    Frame currentFrame = root;
+    sprintf(fileName,"Recording%d.txt",slot);
+    FILE* recording = fopen(fileName,"w");
+    while(currentFrame.next != NULL){
+      fprintf(recording,frameToString(currentFrame));
+    }
+    fclose(recording);
+  	printf("File %s written and closed.\n",fileName);
+  }
+
+  Frame loadRecording(int s){
+    FILE* f = fopen(fileName,"r");
+    char *frame = malloc(50);
+    Frame rt = NULLFRAME;
+    while(fgets(frame,50,f) != NULL){
+      if(rt.analog_main[0] == -1){
+        //rt =
+      }
+    }
+    /*
+    //TEST CODE FOR READING A FILE
+
+    //Initialize 20 char string for read
+    char *a = malloc(sizeof(char) * 10);
+    char *b = malloc(sizeof(char) * 10);
+    char *c = malloc(sizeof(char) * 10);
+    //Open Test file in read mode
+    f = fopen("Test.txt","r");
+    //Read file contents to str
+    fgets(a, 4, f); //string, max chars + 1, file
+    fgets(b, 4, f);
+    fgets(c, 4, f);
+    //Close file
+    fclose(f);
+    printf("%d, %d, %d\n\r",atoi(a),atoi(b),atoi(c));
+    */
+
+    return root; //TODO: placeholder
+  }
+
+  void recordingLoop(){
+    if(root.analog_main[0] == -1){ //checks if it's equal to NULLFRAME
+      root = getCurrentFrame();
+    } else {
+      addFrame(getCurrentFrame());
+    }
+    delay(200);
+  }
+
+  Frame getCurrentFrame(){
+    Frame frame = {
+      {joystickGetAnalog(1,1),joystickGetAnalog(1,2),joystickGetAnalog(1,3),joystickGetAnalog(1,4)},
+      {joystickGetDigital(1,5,JOY_UP),joystickGetDigital(1,5,JOY_DOWN),joystickGetDigital(1,6,JOY_UP),joystickGetDigital(1,6,JOY_DOWN),
+       joystickGetDigital(1,7,JOY_UP),joystickGetDigital(1,7,JOY_DOWN),joystickGetDigital(1,7,JOY_LEFT),joystickGetDigital(1,7,JOY_RIGHT),
+       joystickGetDigital(1,8,JOY_UP),joystickGetDigital(1,8,JOY_DOWN),joystickGetDigital(1,8,JOY_LEFT),joystickGetDigital(1,8,JOY_RIGHT)},
+      {joystickGetAnalog(2,1),joystickGetAnalog(2,2),joystickGetAnalog(2,3),joystickGetAnalog(2,4)},
+      {joystickGetDigital(2,5,JOY_UP),joystickGetDigital(2,5,JOY_DOWN),joystickGetDigital(2,6,JOY_UP),joystickGetDigital(2,6,JOY_DOWN),
+       joystickGetDigital(2,7,JOY_UP),joystickGetDigital(2,7,JOY_DOWN),joystickGetDigital(2,7,JOY_LEFT),joystickGetDigital(2,7,JOY_RIGHT),
+       joystickGetDigital(2,8,JOY_UP),joystickGetDigital(2,8,JOY_DOWN),joystickGetDigital(2,8,JOY_LEFT),joystickGetDigital(2,8,JOY_RIGHT)},
+       NULL,NULL
+    };
+    frame.next=&frame;
+    frame.previous=&frame;
+    return frame;
+  }
+
+  void addFrame(Frame toAdd){
+      toAdd.previous = root.previous;
+      toAdd.next = &root;
+      root.previous = &toAdd;
+  }
+
+  char * frameToString(Frame frame){
+    char * string = malloc(50);
+    sprintf(string,"%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d%d",
+      frame.analog_main[0]+127,
```

Comment:
see comment in header file about macros, these array indeces are not self-explanatory at all